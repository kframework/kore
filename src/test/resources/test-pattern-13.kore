[]

module TEST-PATTERN-13

  axiom{S}
    \rewrites{S,S}(X:S,X:S) []

endmodule []
