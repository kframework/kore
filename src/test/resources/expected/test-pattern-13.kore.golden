Definition
    { definitionAttributes = Attributes []
    , definitionModules =
        [ Module
            { moduleName = ModuleName "TEST-PATTERN-13"
            , moduleSentences =
                [ MetaSentence (SentenceAxiomSentence SentenceAxiom
                    { sentenceAxiomParameters =
                        [ UnifiedObject (SortVariable (Id "S" :: Id Object))
                        ]
                    , sentenceAxiomPattern =
                        Fix (UnifiedPattern (UnifiedObject (Rotate31 (FloorPattern Floor
                            { floorOperandSort =
                                SortVariableSort (SortVariable (Id "S" :: Id Object))
                            , floorResultSort =
                                SortVariableSort (SortVariable (Id "S" :: Id Object))
                            , floorChild =
                                Fix (UnifiedPattern (UnifiedObject (Rotate31 (VariablePattern Variable
                                    { variableName = Id "X" :: Id Object
                                    , variableSort =
                                        SortVariableSort (SortVariable (Id "S" :: Id Object))
                                    }))))
                            }))))
                    , sentenceAxiomAttributes = Attributes []
                    })
                ]
            , moduleAttributes = Attributes []
            }
        ]
    }